### DISCUSSION TITLE
In CI comments how to show projects that have resource changes but no cost change

### DISCUSSION THREAD

**Discussion by aliscott** [infracost/infracost#1159] [Ideas & Feature Requests]
## See https://github.com/infracost/infracost/issues/1538 for the proposed solution

---

The Infracost diff output will show resource changes in a project, even if the overall cost of the project doesn't change. For example, this can happen if a resource is renamed, or a resource moves to a region that has the same prices. 

Current behavior:
* Any project with resource changes is shown in the top summary, even if the resource changes result in a $0 cost change.
* Projects with no resource changes are counted in the `13 projects have no cost estimate changes` message.

Example:
<img width="444" alt="Screenshot 2021-12-01 at 08 15 29" src="https://user-images.githubusercontent.com/158824/144197279-05599aaf-4a9d-40d6-b2fd-b589c60a74e1.png">

I'll post some ideas below, so please update/comment on them and add any of your own.

**Comment by aliscott** [comment]
#### Change the `no cost estimate changes` message

Update the `no cost estimate changes` message to be:
a. `13 projects have no changes`
b. `13 project have no resource changes`

**Reply by alikhajeh1** [reply]
`13 projects have no cost or resource changes` ?

**Comment by aliscott** [comment]
#### Always exclude projects with $0

Don't show any projects with $0 in the top summary, even if they have changes.

**Comment by aliscott** [comment]
#### Add `Resources changed` column

Include a `Resources changed` column in the top summary that shows the number of changed resources, to make it clear why a $0 cost project is there.

**Reply by hugorut** [reply]
This seems like the most complete solution, however, I think without clear indication that this a real pain point - [Change the no cost estimate changes message](https://github.com/infracost/infracost/discussions/1159#discussioncomment-1726780) is the way to go.
